source/development/testing.rst

```diff
@@ -1,6 +1,22 @@
 Testing
 =======
-
+The primary goal of the Bao Test Framework is to provide the infrastructure for
+the testing process of Bao Project components. It encompasses three core
+components: (i) a C code library designed for real-time test handling, (ii) a
+build system based on Nix, and (iii) a Python Tool for comprehensive test
+management.
+
+1. **C Library** - This library comprises a set of utilities that simplify test
+   definition and facilitate result assessment, including streamlined result
+   logging;
+
+2. **Nix Build System** - The Nix Build System introduces an abstraction layer
+   for system builds, enabling the construction of various components, from the
+   firmware to the guests running in each test;
+
+3. **Python Tool** - The Python Tool adds a layer of abstraction to test
+   deployment, managing both the Nix Build System and the C code library
+   components.
```

Comment:
I'd like a bit more of detail. "layer of abstraction" is too abstract. I'd suggest something in the lines of :+1: 
```suggestion
3. **Python Tool** - A command line tool that drives the framework by 
 managing both the Nix Build System and the C code library components
 to build testing artifacts, automating their deployment in the target platform,
 and gathering, analyzing and outputting the final results.
```

Reply:
Agree, thanks! Fixed.